compose/config/fields_schema.json

```diff
@@ -37,7 +37,15 @@
 
         "environment": {
```

Comment:
I didn't use `list_or_dict` for this one as we define some extra constraints in the `patternProperties` for the allowed key names. I was being restrictive cos I was worried about character compatibility for different systems.